src/measure_theory/haar_measure.lean

```diff
@@ -0,0 +1,551 @@
+/-
+Copyright (c) 2020 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import measure_theory.content
+import measure_theory.group
+/-!
+# Haar measure
+
+In this file we prove the existence of Haar measure for a locally compact Hausdorff topological
+group.
+
+## Main Declarations
+
+* `haar_measure`: the Haar measure on a locally compact Hausdorff group. This is a left invariant
+  regular measure. It takes as argument a compact set of the group (with non-empty interior),
+  and is normalized so that the measure of the given set is 1.
+* `haar_measure_self`: the Haar measure is normalized.
+* `is_left_invariant_haar_measure`: the Haar measure is left invariant.
+* `regular_haar_measure`: the Haar measure is a regular measure.
+
```

Comment:
Could you add in the file-level docstring a mathematical sketch of the construction of the Haar measure you are going to implement in the file (maybe giving the names of some formal lemmas alongside the informal explanation)? There are several such constructions, so it would help a lot the readers if they know in which direction you are going.

Reply:
done